backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/FunctionInlining.kt

```diff
@@ -226,9 +223,9 @@ private class Inliner(val globalSubstituteMap: MutableMap<DeclarationDescriptor,
                                       val argumentExpression : IrExpression) {
         val needsEvaluation: Boolean
             get() {
-                if (argumentExpression is IrGetValue) return false                         // Parameter is already GetValue - nothing to evaluate.
```

Comment:
What is the purpose to have such an "optimization"?
For example, are you sure that storing `IrCallableReference` into variable would be less efficient than copying it into every usage?
And actually it is not correct at all:

```
inline fun call(block1: () -> Unit, noinline block2: () -> Int): Int {
    block1()
    return block2()
}

fun main(args: Array<String>) {
    var x = 5
    println(call({ x = 7 }, x::toInt)) 
}
```

Can you provide any case showing that this "optimization" has any effect?